File: 프로그래머스/unrated/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.cs
using System;
using System.Linq;

public class Solution {
    public int solution(int[] arr) 
    {
        int x = 0;
        
        while(true)
        {
            int[] other = new int[arr.Length];
            for(int i = 0; i < arr.Length; ++i)
            {
                int n = arr[i];
                if(n >= 50 && n % 2 == 0)
                    n = n / 2;
                else if(n < 50 && n % 2 == 1)
                    n = n * 2 + 1;

                other[i] = n;            
            }
            
            if(Enumerable.SequenceEqual(arr, other))
                break;
            
            var temp = other;
            other = arr;
            arr = temp;
            ++x;
        }
        
        return x;
    }
}

File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> arr) {
    
    int answer = 0;
    bool anyChange = true;
    while (anyChange)
    {
        anyChange = false;
        
        for (int& num : arr)
        {
            if (num >= 50 && !(num & 1))
            {
                num /= 2;
                anyChange = true;
            }
            else if (num < 50 && (num & 1))
            {
                num = (2 * num) + 1;
                anyChange = true;
            }
        }
        
        ++answer;
    }
    
    return answer - 1;
}